tests/aggregation/tests.py

```diff
@@ -15,6 +16,52 @@
 from .models import Author, Book, Publisher, Store
 
 
+class FilteredAggregateTestCase(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.a1 = Author.objects.create(name='test', age=40)
+        cls.a2 = Author.objects.create(name='test2', age=60)
+
+        cls.a1.friends.add(cls.a2)
+
+    def test_filtered_aggregates(self):
+        agg = Sum('age', filter=Q(name='test2'))
+        self.assertEqual(Author.objects.aggregate(age=agg)['age'], 60)
+
+    def test_double_filtered_aggregates(self):
+        agg = Sum('age', filter=Q(Q(name='test2') & ~Q(name='test')))
+        self.assertEqual(Author.objects.aggregate(age=agg)['age'], 60)
+
+    def test_excluded_aggregates(self):
+        agg = Sum('age', filter=~Q(name='test2'))
+        self.assertEqual(Author.objects.aggregate(age=agg)['age'], 40)
+
+    def test_related_aggregates(self):
+        agg = Sum('friends__age', filter=Q(friends__name='test2'))
+        self.assertEqual(Author.objects.filter(name='test').aggregate(age=agg)['age'], 60)
+
+    def test_case_aggregate(self):
+        agg = Sum(Case(When(friends__age=40, then=F('friends__age'))),
+                  filter=Q(friends__age=40))
+        self.assertEqual(Author.objects.aggregate(age=agg)['age'], 40)
+
+    def test_aggregate_str(self):
+        q = Q(name='test')
+        agg = Sum('test', filter=q)
+        self.assertIn(str(q), str(agg))
+
+    def test_sum_star_exception(self):
+        with self.assertRaisesMessage(ValueError, 'Star cannot be used with filter. Please specify a field.'):
+            Count('*', filter=Q(age=40))
+
+    @unittest.skipUnless(connection.vendor == 'postgresql', "PostgreSQL specific tests")
+    def test_aggregate_multiple_parameters(self):
+        from django.contrib.postgres.aggregates import StringAgg
+        q = Q(name='test')
+        agg = StringAgg(F('name'), delimiter=';', filter=q)
+        self.assertEqual(Author.objects.aggregate(name=agg)['name'], 'test')
```

Comment:
Since `StringAgg` is only implemented on PostgreSQL which uses `FILTER` and not the `CASE(WHEN())` fallback this doesn't actually test the `source_expressions[0]` wrapping. Since we don't have any builtin aggregate with multiple argument this test might not be required after all.